internal/controller/clusterpolicy_controller.go

```diff
@@ -0,0 +1,179 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/policy-meta-operator/internal/utils"
+	edgedbutils "github.com/giantswarm/policy-meta-operator/internal/utils/edgedb"
+)
+
+// PolicyReconciler reconciles a AutomatedException object
+type ClusterPolicyReconciler struct {
+	client.Client
+	EdgeDBClient     *edgedb.Client
+	Scheme           *runtime.Scheme
+	MaxJitterPercent int
+}
+
+var (
+	//GiantSwarm team label
+	GSTeamLabel = "application.giantswarm.io/team"
+	//Policy API Exemption label
+	//TODO: Move to Policy API
+	PolicyAPIExemptionLabel = "policy.giantswarm.io/giantswarm-exempt"
+)
+
+func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	var clusterPolicy kyvernoV1.ClusterPolicy
+
+	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
+		if !errors.IsNotFound(err) {
+			// Error fetching the report
+			log.Log.Error(err, "unable to fetch Kyverno ClusterPolicy")
+		} else {
+			// Policy not found, make sure we don't have it in edgedb either
+			err := edgedbutils.DeleteKyvernoClusterPolicy(ctx, r.EdgeDBClient, req.Name)
+			if err != nil {
+				log.Log.Error(err, "Error deleting ClusterPolicy from database")
+			}
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if clusterPolicy.Spec.ValidationFailureAction.Enforce() {
+		if clusterPolicy.HasValidate() {
+			log.Log.Info("Kyverno ClusterPolicy is in enforce mode and has validating rules, adding it to edgedb")
+
+			policyRuleNames := extractRuleNames(clusterPolicy)
+			policyTargetKinds := extractTargetKinds(clusterPolicy)
+			exempted := isExempted(clusterPolicy)
+
+			if len(policyRuleNames) == 0 || len(policyTargetKinds) == 0 {
+				log.Log.Error(errors.New("Error extracting rule names or target kinds"), fmt.Sprintf("Error extracting rule names or target kinds from Kyverno ClusterPolicy %s", clusterPolicy.Name))
+				return ctrl.Result{}, nil
+			}
+
+			_, err := edgedbutils.InsertKyvernoClusterPolicy(ctx, r.EdgeDBClient, clusterPolicy.Name, policyRuleNames, policyTargetKinds, exempted)
+			if err != nil {
+				log.Log.Error(err, "Error inserting Kyverno ClusterPolicy in database")
+			}
+
+		}
+	}
+
+	return utils.JitterRequeue(utils.DefaultRequeueDuration, r.MaxJitterPercent, log.Log), nil
+}
+
+func isExempted(clusterPolicy kyvernoV1.ClusterPolicy) bool {
+	exempted := true
+
+	// Check if team label exist
+	if _, ok := clusterPolicy.Labels[GSTeamLabel]; ok {
+		// If team label exist, GS workloads are not exempted (they should pass the policy)
+		exempted = false
+	} else {
+		// If team label doesn't exist, make sure we don't have an exemption for this policy
+		if isExempted, ok := clusterPolicy.Labels[PolicyAPIExemptionLabel]; ok {
+			// Use label value to determine if exempted
+			exempted = isExempted == "true"
+		}
+	}
+
```

Comment:
1. The labels aren't mutually exclusive, so we should define the behavior if the team label is set AND the giantswarm-exempt label is set. I'd argue the more specific label (giantswarm-exempt) should take precedence.
2. The logic here is the other way around in most cases (i.e. exempt is the default and we override if it's false)

```suggestion
	}
	
	// Check if the policy has a label enabling or disabling GS exemption.
	if  gsExemptLabelValue, ok := clusterPolicy.Labels[PolicyAPIExemptionLabel]; ok {
	    gsExempt, err = strconv.ParseBool(gsExemptLabelValue)
	    if err != nil {
	        // The label value is garbage. Complain and error out, or default the behavior
	    }
	}
	
	return gsExempt
```